pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -288,18 +285,6 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
 			"--osd-max-object-name-len", "256",
 			"--osd-max-object-namespace-len", "64",
 		)
-
```

Comment:
This removes the place where `crush_location` is set for dir-based OSDs, but the `--crush-location` flag isn't added back to the dir-based OSD to replace this.

Reply:
Do we need to support the crush location for dir-based osds? With support going away for them soon I'd suggest new features like this don't need to apply to the dir-based OSDs.

Reply:
I'm kinda on the same page as @travisn on this one.

Reply:
They aren't deprecated yet, and if we don't have time or resources to deprecate the dir-based OSDs for the next release, we are doing users a disservice here simply to make the argument that adding one line of code back into a PR is too much.

Reply:
@BlaineEXE You're suggesting this code block just be added back since it's already there? It's fine with me, but we should add some comments so it's clear why the location is being set in two different locations. Currently, it's just confusing to see it set two different ways.

Reply:
I was just suggesting that the `--crush-location` flag be added in to replace this removed block. I don't see a need for it to be in the mon config db, but we should still set it somehow for now.